checks/hsts.go

```diff
@@ -0,0 +1,87 @@
+package checks
+
+import (
+	"context"
+	"net/http"
+	"strconv"
+	"strings"
+	"unicode"
+)
+
+type HSTSResponse struct {
+	Message    string `json:"message"`
+	Compatible bool   `json:"compatible"`
+	HSTSHeader string `json:"hstsHeader"`
+}
+
+type Hsts struct {
+	client *http.Client
+}
+
+func NewHsts(client *http.Client) *Hsts {
+	return &Hsts{client: client}
+}
+
+func (h *Hsts) Validate(ctx context.Context, url string) (*HSTSResponse, error) {
+	req, err := http.NewRequestWithContext(ctx, http.MethodHead, url, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := h.client.Do(req)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	hstsHeader := resp.Header.Get("Strict-Transport-Security")
+	if hstsHeader == "" {
+		return &HSTSResponse{Message: "Site does not serve any HSTS headers."}, nil
+	}
+
+	if !strings.Contains(hstsHeader, "max-age") {
```

Comment:
Although max-age is [not optional](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Strict-Transport-Security#directives), I guess we shouldn't assume it'll have max-age in the headers?